src/base/edition.rs

```diff
@@ -703,3 +703,135 @@ unsafe fn extend_rows<N: Scalar>(
         );
     }
 }
+
+/// Extend the number of columns of the `Matrix` with elements from
+/// a given iterator.
+impl<N, R, S> Extend<N> for Matrix<N, R, Dynamic, S>
+where
+    N: Scalar,
+    R: Dim,
+    S: Extend<N>,
+{
+    /// Extend the number of columns of the `Matrix` with elements
+    /// from the given iterator.
+    ///
+    /// # Example
+    /// ```
+    /// # use nalgebra::{DMatrix, Dynamic, Matrix, MatrixMN, Matrix3};
+    ///
+    /// let data = vec![0, 1, 2,      // column 1
+    ///                 3, 4, 5];     // column 2
+    ///
+    /// let mut matrix = DMatrix::from_vec(3, 2, data);
+    ///
+    /// matrix.extend(vec![6, 7, 8]); // column 3
+    ///
+    /// assert!(matrix.eq(&Matrix3::new(0, 3, 6,
+    ///                                 1, 4, 7,
+    ///                                 2, 5, 8)));
+    /// ```
+    ///
+    /// # Panics
+    /// This function panics if the number of elements yielded by the
+    /// given iterator is not a multiple of the number of rows of the
+    /// `Matrix`.
+    ///
+    /// ```should_panic
+    /// # use nalgebra::{DMatrix, Dynamic, MatrixMN};
+    /// let data = vec![0, 1, 2,  // column 1
+    ///                 3, 4, 5]; // column 2
+    ///
+    /// let mut matrix = DMatrix::from_vec(3, 2, data);
+    ///
+    /// // The following panics because the vec length is not a multiple of 3.
+    /// matrix.extend(vec![6, 7, 8, 9]);
+    /// ```
+    fn extend<I: IntoIterator<Item=N>>(&mut self, iter: I) {
+        self.data.extend(iter);
+    }
+}
+
+/// Extend the number of rows of the `Vector` with elements from
+/// a given iterator.
+impl<N, S> Extend<N> for Matrix<N, Dynamic, U1, S>
+where
+    N: Scalar,
+    S: Extend<N>,
+{
+    /// Extend the number of rows of a `Vector` with elements
+    /// from the given iterator.
+    ///
+    /// # Example
+    /// ```
+    /// # use nalgebra::DVector;
+    /// let mut vector = DVector::from_vec(3, vec![0, 1, 2]);
+    /// vector.extend(vec![3, 4, 5]);
+    /// assert!(vector.eq(&DVector::from_vec(6, vec![0, 1, 2, 3, 4, 5])));
+    /// ```
+    fn extend<I: IntoIterator<Item=N>>(&mut self, iter: I) {
+        self.data.extend(iter);
+    }
+}
+
+impl<N, R, S, RV, SV> Extend<Vector<N, RV, SV>> for Matrix<N, R, Dynamic, S>
+where
+    N: Scalar,
+    R: Dim,
+    S: Extend<Vector<N, RV, SV>>,
+    RV: Dim,
+    SV: Storage<N, RV>
```

Comment:
I think it would be useful to ensure that the dimension `R` and `RV` match at compile-time if they a type-level integer. This can be done as follows:

```suggestion
    SV: Storage<N, RV>,
    ShapeConstraint: SameNumberOfRows<R, RV>
```